### PR TITLE
Remove android code / gradle build system / port tests to jUnit / additional tests

### PR DESCRIPTION
**Pull request by alexbirkett**
It would be really awesome if the community had an up-to-date version of cassowary for java. The most recently worked on [fork](https://github.com/ebolwidt/cassowary-java) does not appear to be maintained.

I've removed the android code and added it [to a different fork of this project that deletes all the cassowary code and depends on this project](https://github.com/alexbirkett/pybee-cassowary-java)

I've added support for the gradle build system. This is the default for Android Studio, my hope is that it's straightforward to import the project. You can select the java sdk instead of android if you wish since there are no android dependencies. This means tests can run on your machine using the jre rather than loading an APK on to an Android device, which is quicker! (by could possibly have different behaviour)

I've ported the test cases to jUnit and added some additional tests, some of which fail. I believe the defects will also affect your android project. I need to help to fix them. I'll raise separate issues about them.

I've updated the readme with info about how to depend on this project. Currently the artefacts are hosted on github but maybe we should consider trying to get them into maven central. / jCenter?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,6 @@
 *.egg-info
 *.class
 *.o
-*.jar
```
Comment (freakboy3742) [inline .gitignore]: Why take jar files out of the .gitignore? They're binary artefacts, and shouldn't be stored in version control.

File: gradlew [hunk 2]
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
```
Comment (freakboy3742) [inline gradlew]: So - what's the purpose of this file? It looks like a "let me install Gradle for you" script - does the Gradle community really expect you to do this?

File: org/pybee/android/BoundingBox.java [hunk 3]
```diff
@@ -1,82 +0,0 @@
-package org.pybee.android;
```
Comment (freakboy3742) [inline org/pybee/android/BoundingBox.java]: No problem splitting up the code into two different jar files, but what's the advantage in two different repos? To my mind, it's all Cassowary; as long as the build system makes building individual components optional, I don't see what is  gained by splitting the repo.

File: README.rst [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
 Cassowary is the algorithm that forms the core of the OS X and iOS visual
 layout mechanism.
 
 Quickstart
 ----------
+Gradle users
+
+Add the maven url to your project's ``build.gradle``
 
-To build ``cassowary-java``::
```
Comment (freakboy3742) [inline README.rst]: The quickstart instructions now tell you how to use this in your own project, but now doesn't tell you how to build this project itself.

File: .gitignore (hunk 1 above)
Comment (alexbirkett) [inline .gitignore]: Generally speaking I agree. The exception is [gradle-wrapper.jar](http://www.gradle.org/docs/current/userguide/gradle_wrapper.html)

File: gradlew (hunk 2 above)
Comment (alexbirkett) [inline gradlew]: Exactly. [The wrapper is something you should check into version control. By distributing the wrapper with your project, anyone can work with it without needing to install Gradle beforehand. ](http://www.gradle.org/docs/current/userguide/gradle_wrapper.html)

File: README.rst (hunk 4 above)
Comment (alexbirkett) [inline README.rst]: Good point. Maybe we should add a note about how to build a gradle project.

File: org/pybee/android/BoundingBox.java (hunk 3 above)
Comment (alexbirkett) [inline org/pybee/android/BoundingBox.java]: In my opinion small focused repos containing single projects are cleaner. If somebody wanted to start a Java based cassowary project, they may be put off if they had to download Android code too. Any commits to the Android code would just be noise for them.

**Comment by freakboy3742** [comment]
I've integrated these commits into trunk; I didn't just merge the pull request because I've made a few tweaks along the way. The project is now registered with JCenter; I'm waiting for it to be accepted into the repo.

**Pull request closed without merging**